pkg/controllers/execution/execution_controller.go

```diff
@@ -77,31 +95,74 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 		return controllerruntime.Result{Requeue: true}, err
 	}
 
-	if !work.DeletionTimestamp.IsZero() {
-		// Abort deleting workload if cluster is unready when unjoining cluster, otherwise the unjoin process will be failed.
-		if util.IsClusterReady(&cluster.Status) {
-			err := c.tryDeleteWorkload(clusterName, work)
-			if err != nil {
-				klog.Errorf("Failed to delete work %v, namespace is %v, err is %v", work.Name, work.Namespace, err)
-				return controllerruntime.Result{Requeue: true}, err
-			}
-		} else if cluster.DeletionTimestamp.IsZero() { // cluster is unready, but not terminating
-			return controllerruntime.Result{Requeue: true}, fmt.Errorf("cluster(%s) not ready", cluster.Name)
-		}
-
-		return c.removeFinalizer(work)
-	}
-
 	if !util.IsClusterReady(&cluster.Status) {
 		klog.Errorf("Stop sync work(%s/%s) for cluster(%s) as cluster not ready.", work.Namespace, work.Name, cluster.Name)
 		return controllerruntime.Result{Requeue: true}, fmt.Errorf("cluster(%s) not ready", cluster.Name)
 	}
 
-	return c.syncWork(clusterName, work)
+	resp, err := c.ResourceInformer.BuildResourceInformers(cluster, work, c.eventHandler)
+	if err != nil {
+		return resp, err
+	}
+
+	c.worker.Add(req)
+
+	return controllerruntime.Result{}, nil
+}
+
+func (c *Controller) onUpdate(old, cur interface{}) {
+	unstructuredOld := old.(*unstructured.Unstructured)
+	unstructuredCur := cur.(*unstructured.Unstructured)
+
+	unstructuredOldCopy := unstructuredOld.DeepCopy()
+	unstructuredCurCopy := unstructuredCur.DeepCopy()
+
+	_ = prune.RemoveIrrelevantField(unstructuredOldCopy, prune.RemoveJobTTLSeconds)
+	_ = prune.RemoveIrrelevantField(unstructuredCurCopy, prune.RemoveJobTTLSeconds)
+
+	if !reflect.DeepEqual(unstructuredOldCopy, unstructuredCurCopy) {
+		c.worker.Enqueue(unstructuredCur)
+	}
+}
+
+func (c *Controller) onDelete(obj interface{}) {
+	curObj := obj.(*unstructured.Unstructured)
+
+	c.worker.Enqueue(curObj)
+}
+
+// RunWorkQueue initializes worker and run it, worker will process resource asynchronously.
+func (c *Controller) RunWorkQueue() {
+	workerOptions := util.Options{
+		Name:          "work-execution",
+		KeyFunc:       generateKey,
+		ReconcileFunc: c.syncWork,
+	}
+	c.worker = util.NewAsyncWorker(workerOptions)
+	c.worker.Run(c.ConcurrentWorkSyncs, c.StopChan)
+}
+
+// generateKey generates a key from obj, the key contains cluster, GVK, namespace and name.
+func generateKey(obj interface{}) (util.QueueKey, error) {
+	resource, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("object is not unstructured")
+	}
+
+	workName := util.GetLabelValue(resource.GetLabels(), workv1alpha1.WorkNameLabel)
+	workNamespace := util.GetLabelValue(resource.GetLabels(), workv1alpha1.WorkNamespaceLabel)
+
+	if len(workName) == 0 || len(workNamespace) == 0 {
```

Comment:
`if workName == "" || workNamespace == ""` ?

Reply:
It was referenced from here:  
https://github.com/karmada-io/karmada/blob/d27faeac953043c2274bb8787bbbaaafa7d0dcd6/pkg/controllers/status/work_status_controller.go#L182-L187  

And I have found one more part like this, I will try to change them together.

Reply:
Done.